client/app/src/network/NetworkService.js

```diff
@@ -176,32 +181,42 @@
 
     function getFromPeer(api) {
       var deferred = $q.defer();
-      peer.lastConnection = new Date();
-      $http({
-        url: peer.ip + api,
-        method: 'GET',
-        headers: {
-          'Content-Type': 'application/json',
-          'os': 'ark-desktop',
-          'version': clientVersion,
-          'port': 1,
-          'nethash': network.nethash
-        },
-        timeout: 5000
-      }).then(
-        function(resp) {
-          deferred.resolve(resp.data);
-          peer.isConnected = true;
-          peer.delay = new Date().getTime() - peer.lastConnection.getTime();
-          connection.notify(peer);
-        },
-        function(resp) {
-          deferred.reject("Peer disconnected");
-          peer.isConnected = false;
-          peer.error = resp.statusText || "Peer Timeout after 5s";
-          connection.notify(peer);
+      timeService.getTimestamp().then(
+        function(timestamp) {
+          peer.lastConnection = timestamp;
+          $http({
+            url: peer.ip + api,
+            method: 'GET',
+            headers: {
+              'Content-Type': 'application/json',
+              'os': 'ark-desktop',
+              'version': clientVersion,
+              'port': 1,
+              'nethash': network.nethash
+            },
+            timeout: 5000
+          }).then(
+            function(resp) {
+              timeService.getTimestamp().then(
+                function(success){
+                  deferred.resolve(resp.data);
+                  peer.isConnected = true;
+                  peer.delay = timestamp - peer.lastConnection.getTime();
```

Comment:
success or timestamp? 🤔

Reply:
Should be timestamp, but not needed anymore (See previous comment).